src/backend/distributed/commands/statistics.c

```diff
@@ -0,0 +1,317 @@
+	RangeVar *relation = (RangeVar *) linitial(stmt->relations);
+	Oid relationId = RangeVarGetRelid(relation, AccessExclusiveLock, false);
+
+	if (!IsCitusTable(relationId))
+	{
+		return NIL;
+	}
+
+	EnsureCoordinator();
+
+	QualifyTreeNode((Node *) stmt);
+
+	EnsureSequentialModeForStatisticsDDL();
+
+	ddlJob->targetRelationId = RangeVarGetRelid(relation, AccessExclusiveLock, false);
+	ddlJob->concurrentIndexCmd = false;
+	ddlJob->startNewTransaction = false;
+	ddlJob->commandString = queryString;
+	ddlJob->taskList = DDLTaskList(relationId, queryString);
+
+	List *ddlJobs = list_make1(ddlJob);
+
+	return ddlJobs;
+}
+
+
+/*
+ * GetExplicitStatisticsCommandList returns the list of DDL commands to create
+ * statistics that are explicitly created for the table with relationId. See
+ * comment of GetExplicitStatisticsIdList function.
+ */
+List *
+GetExplicitStatisticsCommandList(Oid relationId)
+{
+	List *createStatisticsCommandList = NIL;
+
+	PushOverrideEmptySearchPath(CurrentMemoryContext);
+
+	List *statisticsIdList = GetExplicitStatisticsIdList(relationId);
+
+	Oid statisticsId = InvalidOid;
+	foreach_oid(statisticsId, statisticsIdList)
+	{
+		char *createStatisticsCommand = pg_get_statisticsobj_worker(statisticsId, true);
+
+		createStatisticsCommandList = lappend(
+			createStatisticsCommandList,
+			makeTableDDLCommandString(createStatisticsCommand));
+	}
+
+	/* revert back to original search_path */
+	PopOverrideSearchPath();
+
+	return createStatisticsCommandList;
+}
+
+
+/*
+ * GetExplicitStatisticsIdList returns a list of OIDs corresponding to the statistics
+ * that are explicitly created on the relation with relationId. That means,
+ * this function discards internal statistics implicitly created by postgres.
+ */
+static List *
+GetExplicitStatisticsIdList(Oid relationId)
+{
+	List *statisticsIdList = NIL;
+
+	Relation pgStatistics = table_open(StatisticExtRelationId, AccessShareLock);
+
+	int scanKeyCount = 1;
+	ScanKeyData scanKey[1];
+
+	ScanKeyInit(&scanKey[0], Anum_pg_statistic_ext_stxrelid,
+				BTEqualStrategyNumber, F_OIDEQ, relationId);
+
+	bool useIndex = true;
+	SysScanDesc scanDescriptor = systable_beginscan(pgStatistics,
+													StatisticExtRelidIndexId,
+													useIndex, NULL, scanKeyCount,
+													scanKey);
+
+	HeapTuple heapTuple = systable_getnext(scanDescriptor);
+	while (HeapTupleIsValid(heapTuple))
+	{
+		Oid statisticsId;
+#if PG_VERSION_NUM >= PG_VERSION_12
+		FormData_pg_statistic_ext *statisticsForm =
+			(FormData_pg_statistic_ext *) GETSTRUCT(heapTuple);
+		statisticsId = statisticsForm->oid;
+#else
+		statisticsId = HeapTupleGetOid(heapTuple);
+#endif
+		statisticsIdList = lappend_oid(statisticsIdList, statisticsId);
+
+		heapTuple = systable_getnext(scanDescriptor);
+	}
+
+	systable_endscan(scanDescriptor);
+	table_close(pgStatistics, NoLock);
+
+	return statisticsIdList;
+}
+
+
+static char *
+pg_get_statisticsobj_worker(Oid statextid, bool missing_ok)
```

Comment:
We might want to add this function to ruleutils instead

Reply:
I thought it would be appropriate to keep it among other functions we use for `CREATE STATISTICS`, all in `statistics.s`. In the future if we need to use this function for some other reason (which I don't expect), we can move it to a more common place like `ruleutils`.

What do you think @halilozanakgul ? Any idea for some another case we need this func?

Reply:
All of the `pg_get_*` functions are in the ruleutils files. I asked to move this too to keep consistent.

Reply:
hmm that makes sense. alright then